src/main/kotlin/it/pagopa/ecommerce/users/services/UserStatisticsService.kt

```diff
@@ -72,4 +72,20 @@ class UserStatisticsService(
             .flatMap { userStatisticsRepository.save(it) }
             .thenReturn(Unit)
     }
+
+    private fun mapUserStatisticsToUserLastPaymentMethodData(
+        lastUsage: LastUsage
+    ): UserLastPaymentMethodData =
+        when (lastUsage.type) {
+            LastUsage.PaymentType.WALLET ->
+                WalletLastUsageData()
+                    .walletId(lastUsage.instrumentId)
+                    .date(lastUsage.date)
+                    .type(lastUsage.type.toString().lowercase(Locale.getDefault()))
```

Comment:
```suggestion
```
remove this setter since it's automatically valued by jackson during serialization.
Openapi  generated class `UserLastPaymentMethodData` have the following jackson annotations:
```code
@JsonIgnoreProperties(
  value = "type", // ignore manually set type, it will be automatically generated by Jackson during serialization
  allowSetters = true // allows the type to be set during deserialization
)
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type", visible = true)
@JsonSubTypes({
  @JsonSubTypes.Type(value = GuestMethodLastUsageData.class, name = "guest"),
  @JsonSubTypes.Type(value = WalletLastUsageData.class, name = "wallet")
})
```